airflow/migrations/versions/f9da662e7089_add_task_log_filename_template_model.py

```diff
@@ -0,0 +1,63 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Add model for task log filename template.
+
+Revision ID: f9da662e7089
+Revises: 5e3ec427fdd3
+Create Date: 2021-12-09 06:11:21.044940
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# Revision identifiers, used by Alembic.
+revision = "f9da662e7089"
+down_revision = "5e3ec427fdd3"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Add model for task log filename template and establish fk on task instance."""
+    op.create_table(
+        "log_filename",
+        sa.Column(sa.Integer, primary_key=True),
+        sa.Column("template", sa.Text, nullable=False),
+        sa.Column("timestamp", sa.UtcDateTime, nullable=False),
+    )
+    with op.batch_alter_table("task_instance") as batch_op:
+        batch_op.add_column(sa.Column("log_filename_id"))
```

Comment:
I think the `server_default` (or equivalent) needs to go here for it to have any effect: https://docs.sqlalchemy.org/en/13/core/defaults.html#server-defaults

> A variant on the SQL expression default is the Column.server_default, which gets placed in the CREATE TABLE statement during a Table.create() operation

Reply:
I switched to use `default` instead. If I read the SQLAlchemy docs correctly, supplying an expression would make it use the expression as an subquery in `INSERT INTO`?